src/appleseed.studio/mainwindow/pythonconsole/pythoneditor.cpp

```diff
@@ -27,50 +27,91 @@
 //
 
 // Interface header.
-#include "pythoninput.h"
+#include "pythoneditor.h"
 
 // appleseed.studio headers.
 #include "pythonhighlighter.h"
+#include "linenumberarea.h"
 
 // Qt headers.
 #include <QKeyEvent>
 #include <QString>
 #include <QStringList>
+#include <QPainter>
 
 // Standard headers.
 #include <string>
 
 namespace appleseed {
 namespace studio {
 
-PythonInput::PythonInput(QWidget* parent)
+PythonEditor::PythonEditor(QWidget* parent)
+  : QPlainTextEdit(parent)
 {
     setUndoRedoEnabled(true);
-    setLineWrapMode(QTextEdit::WidgetWidth);
+    setLineWrapMode(QPlainTextEdit::WidgetWidth);
     setReadOnly(false);
-    setAcceptRichText(false);
     setTextInteractionFlags(
         Qt::TextSelectableByMouse |
         Qt::TextSelectableByKeyboard |
         Qt::TextEditable |
         Qt::TextEditorInteraction);
 
     new PythonSyntaxHighlighter(this->document());
+
+    setStyleSheet("background-color:#303030");
+
+    QFont font("Monospace");
+    font.setStyleHint(QFont::Monospace);
+    setFont(font);
+
+    line_number_area = new LineNumberArea(this);
+
+    connect(this, SIGNAL(cursorPositionChanged()), this, SLOT(slot_highlight_current_line()));
```

Comment:
FYI, there's a variant of connect() to connect a signal and a slot of the same instance:

    connect(this, SIGNAL(cursorPositionChanged()), SLOT(slot_highlight_current_line()));